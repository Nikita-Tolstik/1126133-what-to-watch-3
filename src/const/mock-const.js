export const MockSettings = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  YEAR: 2014
};

export const mockTitles = [

  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`
];
