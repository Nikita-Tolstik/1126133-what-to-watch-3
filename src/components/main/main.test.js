import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main.jsx';
import {MockSettings} from '../../const/mock-const.js';

const mock = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`,
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`,
  },
  {
    img: `img/revenant.jpg`,
    title: `Revenant`,
  },
  {
    img: `img/johnny-english.jpg`,
    title: `Johnny English`,
  },
  {
    img: `img/snatch.jpg`,
    title: `Snatch`,
  },
  {
    img: `img/mindhunter.jpg`,
    title: `Mindhunter`,
  },
  {
    img: `img/war-of-the-worlds.jpg`,
    title: `War of the worlds`,
  },
];

it(`Render Main`, () => {

  const tree = renderer
    .create(<Main
      title={MockSettings.TITLE}
      genre={MockSettings.GENRE}
      year={MockSettings.YEAR}
      films={mock}
      onTitleClick={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
