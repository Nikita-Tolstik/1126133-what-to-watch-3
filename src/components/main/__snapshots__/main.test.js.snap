// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main - authorizationStatus = AUTH and with favoriteFilms 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald2"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link "
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="https://htmlacademy-react-3.appspot.comxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald2"
            height="327"
            src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald2
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              44
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
        onClick={[Function]}
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item false"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Action
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/5"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/6"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/7"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/8"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/8"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
    <div
      className="children-component"
    />
  </div>,
]
`;

exports[`Render Main - authorizationStatus = NO_AUTH and without favoriteFilms 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald2"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link "
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald2"
            height="327"
            src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald2
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              44
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
        onClick={[Function]}
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item false"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Action
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/5"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/6"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/7"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/8"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onClick={[Function]}
            >
              <video
                height="175"
                muted={true}
                poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/8"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald2
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
    <div
      className="children-component"
    />
  </div>,
]
`;
